rest/doc_api.go

```diff
@@ -377,14 +377,20 @@ func (h *handler) handlePutDocReplicator2(docid string, roundTrip bool) (err err
 		parentRev = ifMatch
 	}
 
-	generation, _ := db.ParseRevID(parentRev)
-	generation++
-
 	deleted, _ := h.getOptBoolQuery("deleted", false)
 	newDoc.Deleted = deleted
 
+	var generation int
+	var history []string
+
+	if parentRev != "" {
+		generation, _ = db.ParseRevID(parentRev)
+		history = []string{parentRev}
+	}
+	generation++
```

Comment:
Could you leave this as it was, and just make the change suggested below?  Feels a bit more straightforward.